pkg/http-handler/http-handler_test.go

```diff
@@ -245,50 +246,93 @@ func TestDeleteAppRepository(t *testing.T) {
-			namespaces:   []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
-			expectedCode: 200,
-			headerName:   "X-Consumer-Groups",
-			headerValue:  "nspaces:ns1, nspaces:ns2",
+			name:               "it should return the existing list of namespaces and a 200 when header does not match kubeops arg namespace-header-name",
+			existingNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedCode:       200,
+			additionalHeader:   http.Header{"X-Consumer-Groups": []string{"nspace:ns1", "nspace:ns2"}},
+			namespaceHeaderOptions: kube.KubeOptions{
+				NamespaceHeaderName:    "X-Consumer-Groups",
+				NamespaceHeaderPattern: "namespace:(\\w+)",
+			},
 		},
 		{
-			name:         "it should return the list of namespaces and a 200 when when header does not match kubeops arg ns-header-pattern",
-			namespaces:   []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
-			expectedCode: 200,
-			headerName:   "Y-Consumer-Groups",
-			headerValue:  "namespace:ns1, namespace:ns2",
+			name:               "it should return the existing list of namespaces and a 200 when when header does not match kubeops arg namespace-header-pattern",
+			existingNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedCode:       200,
+			additionalHeader:   http.Header{"Y-Consumer-Groups": []string{"namespace:ns1", "namespace:ns2"}},
+			namespaceHeaderOptions: kube.KubeOptions{
+				NamespaceHeaderName:    "X-Consumer-Groups",
+				NamespaceHeaderPattern: "namespace:(\\w+)",
+			},
+		},
+		{
+			name:               "it should return the existing list of namespaces and a 200 when kubeops arg namespace-header-name is empty",
+			existingNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedCode:       200,
+			additionalHeader:   http.Header{"Y-Consumer-Groups": []string{"namespace:ns1", "namespace:ns2"}},
+			namespaceHeaderOptions: kube.KubeOptions{
+				NamespaceHeaderName:    "",
+				NamespaceHeaderPattern: "namespace:(\\w+)",
+			},
+		},
+		{
+			name:               "it should return the existing list of namespaces and a 200 when kubeops arg namespace-header-pattern is empty",
+			existingNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedNamespaces: []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			expectedCode:       200,
+			additionalHeader:   http.Header{"Y-Consumer-Groups": []string{"namespace:ns1", "namespace:ns2"}},
+			namespaceHeaderOptions: kube.KubeOptions{
+				NamespaceHeaderName:    "X-Consumer-Groups",
+				NamespaceHeaderPattern: "",
+			},
```

Comment:
Perhaps another case: when the header name matches and the header pattern isn't empty but doesn't result in any matches, what happens? (from your code, looks like you only do the early return with the namespaces from header if there are more than 0, so we'd again expect the existingNamespaces in that case)

Reply:
@absoludity Do you think about this test case: https://github.com/kubeapps/kubeapps/pull/2671/files#diff-9a009d80344fc69fac8f2a1d383b6fa3545305bb48ac693d1b894f94ed37853aR282
I added an additional case when some of the header namespaces match the regex : https://github.com/kubeapps/kubeapps/pull/2671/files#diff-9a009d80344fc69fac8f2a1d383b6fa3545305bb48ac693d1b894f94ed37853aR326
The test case shows we need to enclose regex in `^...$`.

Reply:
> @absoludity Do you think about this test case: https://github.com/kubeapps/kubeapps/pull/2671/files#diff-9a009d80344fc69fac8f2a1d383b6fa3545305bb48ac693d1b894f94ed37853aR282

Thanks, perfect (not sure if I missed that as the test name says it's based on the header (itself) not matching, whereas it's actually the case I was looking for, where the regex *value* doesn't match).

> I added an additional case when some of the header namespaces match the regex : https://github.com/kubeapps/kubeapps/pull/2671/files#diff-9a009d80344fc69fac8f2a1d383b6fa3545305bb48ac693d1b894f94ed37853aR326

Great, thanks.

> The test case shows we need to enclose regex in `^...$`.

Excellent, +1 for testing :) (though that's a fix for your actual regex matching use-case, no the code itself, but great to demo).